src/libs/OptionsListUtils.ts

```diff
@@ -2086,6 +2097,96 @@ function formatSectionsFromSearchTerm(
     };
 }
 
+/**
+ * Filters options based on the search input value
+ */
+function filterOptions(options: GetOptions, searchInputValue: string): GetOptions {
+    const searchValue = getSearchValueForPhoneOrEmail(searchInputValue);
+    const searchTerms = searchValue ? searchValue.split(' ') : [];
+
+    // The regex below is used to remove dots only from the local part of the user email (local-part@domain)
+    // so that we can match emails that have dots without explicitly writing the dots (e.g: fistlast@domain will match first.last@domain)
+    const emailRegex = /\.(?=[^\s@]*@)/g;
+
+    const getParticipantsLoginsArray = (item: ReportUtils.OptionData) => {
+        const keys: string[] = [];
+        const visibleChatMemberAccountIDs = item.participantsList ?? [];
+        if (allPersonalDetails) {
+            visibleChatMemberAccountIDs.forEach((participant) => {
+                const login = participant?.login;
+
+                if (participant?.displayName) {
+                    keys.push(participant.displayName);
+                }
+
+                if (login) {
+                    keys.push(login);
+                    keys.push(login.replace(emailRegex, ''));
+                }
+            });
+        }
+
+        return keys;
+    };
+
+    const matchResults = searchTerms.reduceRight((items, term) => {
```

Comment:
I'm a bit confused why we're doing a `reduce` here, is it intentional that we call `filterArrayByMatch` separately for every search term? doesn't think mean we're filtering over all the results iteratively, once for each search term? Wouldn't it be more efficient to filter all the results once, accounting for all the search terms in a single pass?